src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -772,7 +775,75 @@ void AddNativeLibrary (ArchiveFileList files, string path, string abi, string ar
 				Log.LogCodedWarning ("XA4301", path, 0, Properties.Resources.XA4301, item.archivePath);
 				return;
 			}
-			files.Add (item);
+
+			if (!IsEmptyAOTLibrary (item.filePath)) {
+				files.Add (item);
+			} else {
+				Log.LogDebugMessage ($"{item.filePath} is an empty (no executable code) AOT assembly, not including it in the archive");
+			}
+		}
+
+		bool IsEmptyAOTLibrary (string path)
```

Comment:
The only thing I'd suggest is to put this in a new class? `<BuildApk/>` is many lines, could we have a static `ElfHelper.IsEmptyAOTLibrary()` method instead?

Reply:
Sure, it's a good idea, I'll update the code.